simple_salesforce/util.py

```diff
@@ -34,4 +34,14 @@ def date_to_iso8601(date):
 
 class SalesforceError(Exception):
     """Base Salesforce API exception"""
-    pass
+
+    message = 'Unknown error occurred for {url}. Response content: {content}'
+
+    def __init__(self, url, status, resource_name, content):
+        self.url = url
+        self.status = status
+        self.resource_name = resource_name
+        self.content = content
+
+    def __str__(self):
```

Comment:
Hmm... technically `__str__()` should not return `unicode` on Python2, but I'm failing to see where that'd be an issue in this case. So... sure! Thanks for the PR